src/parseSfc.ts

```diff
@@ -26,25 +27,47 @@ export interface FileError extends Error {
   file?: string
 }
 
-export function tryParseCustomBlock(
-  content: string,
-  filePath: string,
-  blockName: string): any {
-  try {
-    return JSON5.parse(content)
+export function tryParseCustomBlock(block: CustomBlock, filePath: string): any {
+  const lang = block.lang ?? 'any'
+  let valid = true
+  let err: Error = new Error()
+  let result: any = {}
+  if (lang === 'json') {
+    try {
+      result = JSON.parse(block.content)
+    }
+    catch (e) {
+      err = e
+      valid = false
+    }
   }
-  catch (err) {
+  else if (lang === 'json5' || lang === 'any') {
     try {
-      return YAML.parse(content)
+      result = JSON5.parse(block.content)
     }
-    catch {
-      const wrapped: FileError = new Error(`Invalid JSON/JSON5/YAML format of <${blockName}> content in ${filePath}\n${err.message}`)
-
-      // Store file path to provide useful information to downstream tools
-      // like friendly-errors-webpack-plugin
-      wrapped.file = filePath
+    catch (e) {
+      err = e
+      valid = false
+    }
+  }
 
-      throw wrapped
+  else if (lang === 'yaml' || (lang === 'any' && !valid)) {
+    try {
+      result = YAML.parse(block.content)
+    }
+    catch (e) {
+      err = e
+      valid = false
     }
   }
```

Comment:
Set default lang to json5 in L31 and change code to:
```ts
if (json) {
  parse json
} elseif (yaml) {
  parse yaml
} elseif (json5) {
  parse json5
} else {
  error
}
```

Reply:
I thought about that, but wanted it to be able to parse YAML without having to specify ```lang="yaml"``` everywhere

Reply:
I think default to Json5 is better. In this case you can add a options to define default parser.

Reply:
OK, I'll add the option and change this.  That's a good solution.  OK if I make "any" an option?  Then we're both happy? :)